src/pages/MyPage.tsx

```diff
@@ -1,24 +1,128 @@
+import { useUserSeriesFetch } from "@/hooks/useUserSeriesFetch";
+import { urlToUserSeries, APIUser } from "@/components/fetchWrapper";
+import ViewdSeries from "@/components/ViewdSeries";
+import DesiredSeries from "@/components/DesiredSeries";
+import { toast } from "react-toastify";
+import { useUpdateSeasonFetch } from "@/hooks/useUpdateSeasonFetch";
+
+interface DataResponseAdd {
+  message: string;
+}
 
 const avatar = require("../images/cat.jpg");
 
 const MyPage = () => {
   const userName = useTypedSelector(userSelectors.userNameSelect);
-  const { loading } = useAuthFetch(sessionStorage.getItem("token"));
+  const userEmail = useTypedSelector(userSelectors.userEmailSelect);
+  useAuthFetch(sessionStorage.getItem("token"));
+
+  const { data, error, loading, seriesFetch } = useUserSeriesFetch(
+    urlToUserSeries,
+    userName
+  );
+
+  const handleClickUpdate = () => {
+    seriesFetch();
+  };
+
+  const [currentIdSeries, setCurrentId] = useState<string | null>(null);
+  const [currentNumberSeason, setCurrentNumber] = useState<number | null>(null);
+
+  useUpdateSeasonFetch(currentIdSeries, currentNumberSeason, userEmail);
+
+  const addSeries = async (
+    id: string,
+    numberSeason: number,
+    paramReq: string
+  ) => {
+    const dataResponse: DataResponseAdd = await APIUser.addedSeriesToList(
+      paramReq,
+      {
+        email: userEmail as string,
+        idseries: id,
+        numberseason: numberSeason,
+      }
+    );
+    if (dataResponse) {
+      toast.success(dataResponse.message);
+    }
+    return Promise.resolve();
+  };
+
+  const deleteSeries = async (id: string, paramReq: string) => {
+    const dataResponse: DataResponseAdd = await APIUser.deleteSeriesFromList(
+      paramReq,
+      {
+        email: userEmail as string,
+        idseries: id,
+      }
+    );
+    if (dataResponse) {
+      toast.success(dataResponse.message);
+    }
+    return Promise.resolve();
+  };
+
+  const updateSeason = (id: string, numberSeason: number) => {
+    setCurrentId(id);
+    setCurrentNumber(numberSeason);
+  };
+
   return (
     <React.Fragment>
-      {loading ? (
+      {error && <div className="errorDiv">Упс... Что-то пошло не так</div>}
+      {loading && (
         <Spinner animation="border" variant="light" className="spiner" />
-      ) : (
-        <div className="mainContainerMyPage">
-          <img src={avatar} alt="avatarUser" className="avatarUser" />
-          {userName}
-        </div>
+      )}
+      {data && (
+        <Container>
+          <Row className="mainContainerMyPage">
+            <Col>
+              <img src={avatar} alt="avatarUser" />
+              {userName}
+            </Col>
+          </Row>
+          <Row>
+            <Tabs defaultActiveKey={"viewd"} id="uncontrolled-tab-example">
+              <Tab eventKey={"viewd"} title="Просмотренные">
+                {data.bwseries.map((el) => {
+                  if (el.bwlistsusers.viewdseries) {
```

Comment:
Let's filter data.bwseries array by viewdseries before the rendering.

```suggestion
                  viewdseries .map((el)
```